include/tudocomp_stat/StatPhase.hpp

```diff
@@ -126,42 +158,53 @@ class StatPhase {
 
         m_parent = s_current;
 
-        m_data = std::make_unique<PhaseData>();
-        m_data->title(std::move(title));
+        m_title = std::move(title);
+        m_sub = json::array();
+        m_stats = json();
+
+        // initialize extensions
+        for(auto ctor : m_extension_registry) {
+            m_extensions.emplace_back(ctor());
```

Comment:
Hm, `emplace_back()` so zu benutzten führt etwas am Ziel vorbei. Wenn man nur das vollständige Object übergeben will hat die Methode effektiv die selbe Semantik wie `push_back()`.

Reply:
Die idee war, dass der `unique_ptr` in `m_extensions` dann direkt mit rvalue-Referenz konstruiert wird, so dass ich kein `std::move` dahinsetzen muss. Das wäre bei `push_back` notwendig, oder?

Reply:
OK.

Aber was führt denn da jetzt "am Ziel vorbei"? In beiden Fällen passiert das, was ich erreichen wollte: das neu erzeugte Objekt wird per "move" ans Ende angefügt. Im Falle von `emplace_back` über den Move-Konstruktor, im Falle von `push_back` vermutlich durch ein move assignment. Gibt es bei irgendeiner Alternative ein nennenswertes potentielles Problem?

Reply:
Ah, ich meinte damit das `emplace_back` dafür existiert um eine Objekt direkt in der Datenstruktur zu instantiieren. Wenn man es benutzt um nur den Move Konstruktor aufzurufen ist das unnötige Komplexität im Vergleich dazu das ansonsten mehr gebräuchlichen `push_back` zu benutzen.

Stimmt aber, es ist weder falsch noch langsamer.